client/signup/steps/design-picker/preview-toolbar.jsx

```diff
@@ -15,18 +15,34 @@ const DesignPickerPreviewToolbar = ( {
 	showDeviceSwitcher,
 	setDeviceViewport,
 	translate,
+	showSiteAddressBar,
+	devicesToShow,
 } ) => {
 	const devices = React.useRef( {
 		computer: { title: translate( 'Desktop' ), icon: computer, iconSize: 36 },
 		tablet: { title: translate( 'Tablet' ), icon: tablet, iconSize: 24 },
 		phone: { title: translate( 'Phone' ), icon: phone, iconSize: 24 },
 	} );
 
+	function filterDevices( selectedDevices ) {
+		// If invalid inputs, display all possible devices
+		// If at least one valid input, Will filter out any devices that do not exist in the 'possibleDevices' array above
+		let filteredPossibleDevices = [];
+		if ( Array.isArray( selectedDevices ) && selectedDevices.length > 0 ) {
+			filteredPossibleDevices = selectedDevices.filter( ( device ) => {
+				return possibleDevices.includes( device );
+			} );
+		}
+		return filteredPossibleDevices.length === 0 ? possibleDevices : filteredPossibleDevices;
+	}
+
+	const filteredPossibleDevices = filterDevices( devicesToShow );
```

Comment:
I wonder if its worth using `useMemo` here? 🤔  Maybe these calculations are too simple to warrant its overhead though.

Reply:
Hmmm I can take note of this and look into it.

Reply:
> Although my guess is these calculations are too simple to warrant its overhead anyways.

Thought about this as well, but came to the same conclusion about overhead. I'd wager it will have virtually no effect on actual performance.

Reply:
> I'd wager it will have virtually no effect on actual performance.

No effect or even possibly negative! I think I overused them when I first found out about it. 😆